lms/djangoapps/courseware/grades.py

```diff
@@ -392,7 +392,7 @@ def _grade(student, course, keep_raw_scores):
 
     More information on the format is in the docstring for CourseGrader.
     """
-    course_structure = get_course_blocks(student, course.location)
+    course_structure = getattr(course, 'structure', [])
```

Comment:
Answered my own question thanks to tests - there are alternate ways to hit these methods, so we can't assume that `course.structure` exists.